src/charm.py

```diff
@@ -58,9 +78,44 @@ def _on_peer_relation_changed(self, event):
             ] = event.relation.data[event.app]["MYSQL_ROOT_PASSWORD"]
             logger.info("Storing MYSQL_ROOT_PASSWORD in StoredState")
 
-    def _on_config_changed(self, _):
-        """This method handles the .on.config_changed() event"""
-        self._configure_pod()
+    def _on_config_changed(self, event):
+        """Set a new Juju pod specification"""
+        self.unit.status = WaitingStatus("Setting up containers.")
+        needs_restart = False
+        container = self.unit.get_container(PEER)
+        plan = container.get_plan().to_dict()
```

Comment:
Could replace this with 

`services = container.get_plan().to_dict().get("services", {})`

Which could tidy up below (see next comment)

Reply:
nice catch! Done.